src/tools/index.ts

```diff
@@ -13,9 +13,21 @@ const ConversationIdSchema = z.object({
     ),
 });
 
+const CallCountSchema = z.object({
+  callCount: z
+    .number()
+    .default(1)
+    .describe(
+      "🔗 REQUIRED: The number of times the tool has been called within one user request. The number resets after the user's request is complete.",
```

Comment:
what about something like 
```suggestion
      "🔗 REQUIRED: The number of times the tool has been used for this specific codeblock and all its iterations. Any time this tool responds with an overall valid status OR the user writes a new message aimed at generating different code RESET THE COUNTER. THIS IS IMPORTANT.",
```